site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -0,0 +1,217 @@
+                    )
+                );
+            }
+            else{
+                $oh_queue = $this->core->getQueries()->getInstructorQueue();
+                return Response::WebOnlyResponse(
+                    new WebResponse(
+                        'OfficeHoursQueue',
+                        'showQueueInstructor',
+                        $oh_queue,
+                        $this->core->getConfig()->getCourse()
+                        )
+                    );
+                }
+            }
+    /**
+    * @param
+    * @Route("/{_semester}/{_course}/office_hours_queue/add", methods={"POST"})
+    * @return Response
+    */
+    public function addPerson(){
+        if(!isset($_POST['code']) || !isset($_POST['name'])){
+            $this->core->addErrorMessage("Missing name or code in request");
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+            );
+        }
+        $section_id = $this->core->getQueries()->isValidCode($_POST['code']);
+        if($_POST['name'] !== "" && !is_null($section_id) && $this->core->getQueries()->isQueueOpen()){
+            //Add the user to the database
+            $oh_queue = $this->core->getQueries()->getQueueByUser($this->core->getUser()->getId());
+            if(!$oh_queue->isInQueue()){
+                if($this->core->getQueries()->addUserToQueue($this->core->getUser()->getId(), $_POST['name'])){
+                    $this->core->addSuccessMessage("Added to queue");
+                }
+                else{
+                    $this->core->addErrorMessage("Unable to add to queue");
+                }
+
+            }
+            else{
+                $this->core->addErrorMessage("You are already in the queue");
+            }
+        }else{
+            if($_POST['name'] == ""){
+                $this->core->addErrorMessage("Invalid Name");
+            }
+            else if(is_null($section_id)){
+                $this->core->addErrorMessage("Invalid Code");
+            }
+            else if(!$this->core->getQueries()->isQueueOpen()){
+                $this->core->addErrorMessage("Queue is closed");
+            }
+
+        }
+        return Response::RedirectOnlyResponse(
+            new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+        );
+    }
+
+    /**
+    * @param
+    * @Route("/{_semester}/{_course}/office_hours_queue/startHelp", methods={"POST"})
+    * @AccessControl(role="LIMITED_ACCESS_GRADER")
+    * @return Response
+    */
+    public function startHelpPerson(){
+        if(!isset($_POST['entry_id'])){
+            $this->core->addErrorMessage("Missing entry ID");
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+            );
+        }
+        if($this->core->getUser()->accessGrading())
```

Comment:
You should be able to get rid of this if statement check as that should be self-contained within the `@AccessControl` annotation test.